web/text_layer_builder.js

```diff
@@ -181,26 +181,25 @@ class TextLayerBuilder {
    * clicked. This reduces flickering of the content if the mouse is slowly
    * dragged up or down.
    */
-  #bindMouse() {
+  #bindMouse(end) {
     const { div } = this;
 
+    let isFirefox;
```

Comment:
Please define this conditionally instead, to reduce duplication below.
```suggestion
    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
      // eslint-disable-next-line no-var
      var isFirefox =
        (typeof PDFJSDev === "undefined" || !PDFJSDev.test("CHROME")) &&
        getComputedStyle(end).getPropertyValue("-moz-user-select") === "none";
      }
```

Reply:
This needs to be done lazily, because initially `getComputedStyle(end).getPropertyValue("-moz-user-select")` returns the empty string even on Firefox. If you prefer, I can wrap it in a cached function to avoid repetition:
```js
    if (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) {
      // eslint-disable-next-line no-var
      var isFirefox = () => {
        const res = (typeof PDFJSDev === "undefined" || !PDFJSDev.test("CHROME")) &&
          getComputedStyle(end).getPropertyValue("-moz-user-select") === "none";
        isFirefox = () => res;
        return res;
      }
    }
```
(I'm caching it because `getComputedStyle` triggers layout recomputation, and `selectionchange` could potentially be fired very frequently)